frontend/src/components/VisitRouter/index.js

```diff
@@ -38,13 +40,33 @@ const VisitRouter = observer(() => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const handleSubmitVisit = () => {
+  const handleSubmitVisit = async () => {
     // TODO: logic protecting against concurrent queue entries
     // if existing visit we are coming from QueueTable, so update visit
-    if (existingVisit.id) {
-      participantStore.updateVisit()
-    } else {
-      participantStore.createVisit()
+    try {
+      let isValid = await validateVisitForm(existingVisit)
+      if (existingVisit.id) {
+        !isValid.length
+          ? participantStore.updateVisit()
+          : isValid.map(error =>
+              participantStore.setSnackbarState(
+                `Error location: ${error.name}, Error message: ${error.message}`
+              )
+            )
+      } else {
+        !isValid.length
+          ? participantStore.createVisit()
+          : isValid.map(error =>
+              console.log(
```

Comment:
is this console log supposed to be here?

Reply:
No sirree!